modules/ocf/functions/lookup_wrap.pp

```diff
@@ -0,0 +1,6 @@
+function ocf::lookup_wrap($item) {
+  $item ? {
+    String  => lookup($item),
```

Comment:
I'm a little concerned that this doesn't allow for string literals, because I feel like those would be pretty useful to be able to use. Maybe use `lookup()` with a fallback of the item itself instead, so like `lookup($item, {'default_value': $item})`? Also, arrays can be passed as the argument to `lookup`, so leaving open support for that would be cool.

My main concern with the alternative I've mentioned is that it would be easy to remove some thing from hiera and have it fail silently, so there's some balance to be struck here between usability (actually being able to use strings) and reliability/predictability (not having this silently break with hiera changes).

Thoughts?

Reply:
Actually, the regex you had here before was decent, because then you can try to check if it's a hiera-like value or not. Maybe we can make a custom type ([type alias](https://puppet.com/docs/puppet/5.5/lang_type_aliases.html) for a regex) or something? That seems like it could be pretty neat

Reply:
I am -1 on guessing whether an `$item` is supposed to be a hiera password or literal string. It seems like there would be too many edge cases. Better would be to have the user call `lookup` manually.

Reply:
I'd agree on that, guessing seems like not the way to go, and adding `lookup` when needed seems much simpler. Then this wrapper wouldn't even be needed either, which I think would be nicer overall.

Reply:
Well it'd still be helpful to have a wrapper to verify password validity nonetheless